sopel/irc/__init__.py

```diff
@@ -330,12 +343,21 @@ def on_error(self):
         self.last_error_timestamp = datetime.utcnow()
         self.error_count = self.error_count + 1
 
-    def change_current_nick(self, new_nick):
+    def rebuild_nick(self) -> None:
+        """Rebuild nick as a new identifier.
+
+        This method exists to update the casemapping rules for the
+        :class:`~sopel.tools.identifiers.Identifier` that represents the bot's
+        nick.
```

Comment:
```suggestion
        nick, e.g. after ISUPPORT info is received.
```

Good idea to mention _why_ one might want to use this? 😸 Can link to some relevant doc section about ISUPPORT.

I also wonder if, perhaps, the method name should be `_rebuild_nick` since it's not really intended for public use?

Reply:
I was thinking about having that private. However, I also realized that a plugin may want to override coretasks's behavior, or have some way to manage the bot's nick and its rules. This method is idempotent, there are very few risk of anything bad happening (if any), and it doesn't accept any parameter, reducing further the risk of anything wrong done to the bot.

Reply:
True enough, I suppose. My only real argument against making it public is that it slightly clutters the docs for new users trying to grok the basics—but then, that's what the other current project (expanding the prose-style docs about writing plugins) is meant to address.

Reply:
> that's what the other current project (expanding the prose-style docs about writing plugins) is meant to address.

Exactly yes!